public/components/reindex.react.js

```diff
@@ -33,6 +39,11 @@ export default class ReindexComponent extends React.Component {
         });
     }
 
+    onChildChanged(newState) {
+        this.setState({ editModeOn: newState });
```

Comment:
This edit mode could maybe also stored in the router, if it's 123/edit then edit mode is on, then no need to manage it in a component. The other components can then just trigger history changes. I'm also a little unsure of the naming of `onChildChanged` maybe something like updateEditModeState would be clearer.